src/Isotope80/Log.cs

```diff
@@ -4,93 +4,160 @@
             Children = children;
         }
 
-        public Node With(string message = null, Log children = null) =>
-            new Node(message ?? Message, children ?? Children);
-
-        public Node Append(string message, Action<string, int> action, int depth) => 
-            With(children: Children.Append(message, action, depth+1));
-
-        public Node Push(string message, Action<string, int> action, int depth) => 
-            With(children: Children.Push(message, action, depth+1));
-
-        public Node Pop() => With(children: Children.Pop());
-
-        public static Node New(string message) => new Node(message, Log.Empty);
-
-        public Seq<string> ToString(int indent) =>
-            Seq1(new string('\t', indent) + Message)
-                .Append(Children.ToString(++indent));
+        /// <summary>
+        /// Add a message to the log
+        /// </summary>
+        /// <param name="ctx">Context</param>
+        public (Log Parent, Log Child) Context(string ctx)
+        {
+            var child = new Log(Indent + 1, LogType.Info, ctx, default);
+            return (new Log(Indent, Type, Message, Children.Add(child)), child);
+        }
 
-        public Seq<string> Trace(int indent) =>
-            Seq1(new string('\t', indent) + Message)
-               .Append(Children.Trace(++indent));
+        /// <summary>
+        /// Add a log entry
+        /// </summary>
+        public Log Add(Log log) =>
+            new Log(Indent, Type, Message, Children.Add(log));
+
+        /// <summary>
+        /// Add a message to the log
+        /// </summary>
+        /// <param name="message">Message to log</param>
+        public Log Info(string message) =>
+            new Log(Indent, Type, Message, Children.Add(new Log(Indent + 1, LogType.Info, $"INFO: {message}", default))); 
+
+        /// <summary>
+        /// Add a message to the log
+        /// </summary>
+        /// <param name="message">Message to log</param>
+        public Log Warning(string message) =>
+            new Log(Indent, Type, Message, Children.Add(new Log(Indent + 1, LogType.Warn, $"WARN: {message}", default)));
+
+        /// <summary>
+        /// Add a message to the log
+        /// </summary>
+        /// <param name="message">Message to log</param>
+        public Log Error(string message) =>
+            new Log(Indent, Type, Message, Children.Add(new Log(Indent + 1, LogType.Error, $"ERRO: {message}", default)));
+
+        /// <summary>
+        /// Create a new Node
+        /// </summary>
+        public static Log New(string message) => 
+            new Log(0, LogType.Info, message, default);
+
+        /// <summary>
+        /// ToString
+        /// </summary>
+        public override string ToString() =>
+            String.Join(Environment.NewLine, ToSeq());
+
+        /// <summary>
+        /// ToSeq
+        /// </summary>
+        public Seq<string> ToSeq() =>
+            Seq1(new string('\t', Indent) + Message)
```

Comment:
Since the `Empty` has `Indent` set to `-1` this `ToSeq` will always throw because the repetition has to be non-negative.

Reply:
Good spot.  Sorted now 👍